### PR TITLE
GPU and Accelerator Model and Manufacturer Support w/ free-tier, auto-recovery, dedicated-hosts, and cpu-manufacturer

### PR DESCRIPTION
**Pull request by bwagner5**
Issue #, if available:
N/A

Description of changes:
 - Upgrade to go 1.18
 - Parallelize filter preparation and execution
 - Add new filters:
   - auto-recovery
   - free-tier
   - dedicated-hosts
   - cpu-manufacturer
   - gpu-manufacturer
   - gpu-model
   - inference-acceleartor-manufacturer
   - inference-accelerator-model

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/selector/comparators.go [hunk 1]
```diff
@@ -219,6 +265,16 @@ func getEBSOptimizedBaselineIOPS(ebsInfo *ec2.EbsInfo) *int64 {
 	return ebsInfo.EbsOptimizedInfo.BaselineIops
 }
 
+func getCPUManufacturer(instanceTypeInfo *ec2.InstanceTypeInfo) *string {
+	if contains(instanceTypeInfo.ProcessorInfo.SupportedArchitectures, ec2.ArchitectureTypeArm64) {
+		return aws.String("aws")
+	}
+	if amdRegex.Match([]byte(*instanceTypeInfo.InstanceType)) {
```
Comment (snay2) [inline pkg/selector/comparators.go:272]: Regex matching the instance type seems fragile with all the possible current and future instance type names. Is that really the best (or only) way to suss out if the CPU is AMD?

File: pkg/selector/comparators.go [hunk 2]
```diff
@@ -140,6 +142,50 @@ func getTotalGpuMemory(gpusInfo *ec2.GpuInfo) *int64 {
...
+	return manufacturers
+}
+
+func getInferenceAcceleratorModels(acceleratorInfo *ec2.InferenceAcceleratorInfo) []*string {
+	if acceleratorInfo == nil {
+		return nil
+	}
+	var models []*string
+	for _, info := range acceleratorInfo.Accelerators {
+		models = append(models, info.Manufacturer)
```
Comment (snay2) [inline pkg/selector/comparators.go:184]: This looks to be adding `.Manufacturer` to the `models` array, but that's also what `getInferenceAcceleratorManufacturers()` is doing. Is that correct?

File: pkg/selector/comparators.go (hunk 1 above)
Comment (bwagner5) [reply pkg/selector/comparators.go:272]: It is currently the only way to tell if a node is AMD :(  But it's well documented in instance type names https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#:~:text=capabilities%2C%20such%20as%3A-,a%20%E2%80%93%20AMD%20processors,-g%20%E2%80%93%20AWS%20Graviton

File: pkg/selector/comparators.go (hunk 2 above)
Comment (bwagner5) [reply pkg/selector/comparators.go:184]: oooooooooops

Comment (bwagner5) [reply pkg/selector/comparators.go]: fixed

**Review by snay2 — APPROVED** [review]

**Pull request merged**